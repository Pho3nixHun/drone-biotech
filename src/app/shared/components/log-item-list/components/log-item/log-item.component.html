<div class="flex flex-col px-2 gap-y-2">
    <div class="flex flex-row justify-between">
        <ng-content select="[strong, time]" />
    </div>
    <ng-content select="p" />
</div>
