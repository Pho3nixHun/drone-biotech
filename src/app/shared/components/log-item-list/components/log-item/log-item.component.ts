import { Component } from '@angular/core';

@Component({
    selector: 'app-log-item',
    imports: [],
    templateUrl: './log-item.component.html',
})
export class LogItemComponent {}
