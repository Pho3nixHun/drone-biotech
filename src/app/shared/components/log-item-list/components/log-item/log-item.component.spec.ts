import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LogItemComponent } from './log-item.component';
import { Component } from '@angular/core';
@Component({
    imports: [LogItemComponent],
    template: `
        <app-log-item>
            <strong></strong>
            <time datetime=""></time>
            <p></p>
            <div>Should not be projected</div>
        </app-log-item>
    `,
})
class TestHostComponent {}
describe('LogItemComponent', () => {
    let compiled: HTMLElement;
    let fixture: ComponentFixture<TestHostComponent>;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
