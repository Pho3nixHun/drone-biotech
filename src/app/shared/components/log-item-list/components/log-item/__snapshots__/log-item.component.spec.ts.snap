// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogItemComponent should render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-log-item>
    <div
      class="flex flex-col px-2 gap-y-2"
    >
      <div
        class="flex flex-row justify-between"
      >
        <strong />
        <time
          datetime=""
        />
      </div>
      <p />
    </div>
  </app-log-item>
</div>
`;
